### ISSUE TITLE
Cannot use empty key for Dictionary

### ISSUE THREAD

**Issue by codeinabox** [neomake/neomake#984]
When opening CSS files I get the error "Cannot use empty key for Dictionary"

### Output of the ":verb NeomakeInfo" command

> #### Neomake debug information
> Async support: 1
> Current filetype: css
> 
> ##### Enabled makers
> For the current filetype (with :Neomake):
> Error detected while processing function neomake#DisplayInfo[8]..<SNR>116_display_maker_info[1]..neomake#GetEnabledMakers[33]..neomake#utils#AvailableMakers[1]..neomake#utils#MakerIsAvailable:
> line   11:
> E713: Cannot use empty key for Dictionary
>

**Comment by blueyed** [comment]
Apparently your `maker.exe` is empty: https://github.com/neomake/neomake/blob/1cc8a566a3fc5899e057e6eee2f206a443bb7318/autoload/neomake/utils.vim#L212.
Please elaborate on your config.

Is the above all the output from `:verb NeomakeInfo`?!

**Comment by codeinabox** [comment]
Yes. I only seem to have this issue with CSS files for now.

**Comment by blueyed** [comment]
Well, that's not enough information for anyone to help you there.
Do you have custom Neomake config?  (It should show up in `:NeomakeInfo`, and the issue template provides a snippet to copy'n'paste it, which you have not done?!)

**Comment by codeinabox** [comment]
But that's the thing, when I run NeomakeInfo it results in that error. I can output my config from another file type say vim but for CSS I always get that error

> #### Neomake debug information
> Async support: 1
> Current filetype: vim
> 
> ##### Enabled makers
> For the current filetype (with :Neomake): None.
> NOTE: you can define g:neomake_vim_enabled_makers to configure it (or b:neomake_vim_enabled_makers).
> 
> For the project (with :Neomake!): None.
> NOTE: you can define g:neomake_enabled_makers to configure it.
> 
> Default maker settings:
>  - remove_invalid_entries: 0
>  - buffer_output: 1
> 
> ##### Settings
> ```
> g:neomake_php_phpmd_extra_args = 'cleancode,codesize,controversial,design,naming,unusedcode'
> g:neomake_javascript_enabled_makers = ['eslint']
> g:neomake_place_signs = 1
> g:neomake_php_phpcs_extra_args = '--standard=PSR2'
> g:neomake_jsx_enabled_makers = ['eslint']
> 
> shell: /usr/local/bin/bash
> shellcmdflag: -c
> Windows: 0
> ```
>

**Comment by blueyed** [comment]
> But that's the thing, when I run NeomakeInfo it results in that error.

Oh!  That's bad.. ;)  (I did not get that part)

Please try the following patch to get more info:

```diff
diff --git i/autoload/neomake/utils.vim w/autoload/neomake/utils.vim
index 70143c0..971b53a 100644
--- i/autoload/neomake/utils.vim
+++ w/autoload/neomake/utils.vim
@@ -205,6 +205,7 @@ function! neomake#utils#MakerIsAvailable(ft, maker_name) abort
         return 1
     endif
     let maker = neomake#GetMaker(a:maker_name, a:ft)
+    echom string(a:maker_name) string(maker)
     if empty(maker)
         return 0
     endif
```

Do you have `csslint` and/or `stylelint` installed?

**Comment by codeinabox** [comment]
Neither `csslint` or `stylelint` is installed

**Comment by codeinabox** [comment]
In the interim adding this to my .vimrc gets stops the error

```viml
let g:neomake_css_enabled_makers = []
```

**Comment by blueyed** [comment]
@codeinabox 
The output from what I posted above would be helpful.
In that case `maker` should be empty (and you should not run into the error).

**Comment by codeinabox** [comment]
> #### Neomake debug information
> Async support: 1
> Current filetype: css
> 
> ##### Enabled makers
> For the current filetype (with :Neomake): None.
> NOTE: you can define g:neomake_css_enabled_makers to configure it (or b:neomake_css_enabled_makers).
> 
> For the project (with :Neomake!): None.
> NOTE: you can define g:neomake_enabled_makers to configure it.
> 
> Default maker settings:
>  - remove_invalid_entries: 0
>  - buffer_output: 1
> 
> ##### Settings
> ```
> g:neomake_javascript_enabled_makers = ['eslint']
> g:neomake_place_signs = 1
> g:neomake_css_enabled_makers = []
> g:neomake_jsx_enabled_makers = ['eslint']
> 
> shell: /usr/local/bin/bash
> shellcmdflag: -c
> Windows: 0
> ```
>

**Comment by blueyed** [comment]
I've meant the diff/patch/echom.

**Comment by blueyed** [comment]
Might be fixed by now, or with https://github.com/neomake/neomake/pull/1131 landing shortly.
Hard to tell without the requested feedback.
Please create a new issue if it is still happening (after #1131).
